csrc/val_graph_visitor.cpp

```diff
@@ -151,4 +151,27 @@ void ValGraphVisitor::traverse() {
   }
 }
 
+ValGroups ValGraphBFS::getReachableValsFrom(
```

Comment:
IIUC, this function is filtering out all the reachable `vals` from the input para `vals`. `Reachable` means traverse from `from` to `vals`. The function name makes me confusing, I was thinking about getting all the reachable values from `from`. Maybe add comment or revise the name.